### PR TITLE
feat: add a cloudflare-streaming wrapper

### PR DESCRIPTION
**Pull request by vicb**
@conico974 Do you think "cloudflare-streaming" is a reasonable name or "cloudflare-server" would be better (or any other idea)?

The changes are somehow tested - might require minor tweaks later.

I have spread the changes across different commits per unit of work to ease the review.

Thanks!

### CODE REVIEW

**Review by conico974 — COMMENTED** [review]
I think it would make more sense to use `cloudflare-server` instead of `cloudflare-streaming`.
The other one support streaming as well, it's just that it only works with next `edge` runtime.

Even better we could rename the other one `cloudflare-edge` and this one `cloudflare-node` which would allow people to understand more easily which one to use depending on the runtime they use

**Comment by vicb** [comment]
> Even better we could rename the other one `cloudflare-edge` and this one `cloudflare-node` which would allow people to understand more easily which one to use depending on the runtime they use

Great idea. I'll do that (and alias the former "cloudflare" to "cloudflare-edge" for BC!

File: packages/open-next/src/overrides/wrappers/cloudflare-streaming.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  InternalEvent,
+  InternalResult | ({ type: "middleware" } & MiddlewareOutputEvent)
+> =
+  async (handler, converter) =>
+  async (
+    request: Request,
+    env: Record<string, string>,
+    ctx: any,
+  ): Promise<Response> => {
+    globalThis.process = process;
```
[the remark this answers is not in the document]
Comment (vicb) [reply packages/open-next/src/overrides/wrappers/cloudflare-streaming.ts:18]: `process` is a global in node compat mode so I think this should be dropped entirely.

File: packages/open-next/src/overrides/wrappers/cloudflare-streaming.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    ctx.waitUntil(handler(internalEvent, streamCreator));
+
+    return promiseResponse;
+  };
+
+export default {
+  wrapper: handler,
+  name: "cloudflare-streaming",
+  supportStreaming: true,
+  edgeRuntime: true,
```
[the remark this answers is not in the document]
Comment (vicb) [reply packages/open-next/src/overrides/wrappers/cloudflare-streaming.ts:76]: I added this value without paying too much attention but I agree it should be false.

**Comment by vicb** [comment]
@conico974 do you think we could merge this PR as is and I would address the comments in a follow up early next week. The rationale is that I have a pending https://github.com/opennextjs/opennextjs-cloudflare/pull/140 using this and I think if I push here it might update the pre-release and break that PR (not sure if the version changes or not).

**Review by conico974 — APPROVED** [review]
Yeah we can merge it's fine. Thanks

**Pull request merged**